### ISSUE TITLE
Can't access predictions 

### ISSUE THREAD

**Issue by lmd-santos** [tensorflow/tensorflow#14381]
------------------------

### System information

- OS Platform and Distribution: Windows 10
- TensorFlow installed from: binary
- TensorFlow version: Tensorflow 1.4.0
- Python version: Python 3.6
- CUDA/cuDNN version: CUDA 8 / cuDNN 6
- GPU model and memory: GTX M950

### Describe the problem
When trying to print the predictions from DNNClassifier class i only get "<generator object Estimator.predict at 0x000001AFE1E24EB8>". I used the exact code written in the Tensorflow tutorials.

### Source code / logs
train_input_fn = tf.estimator.inputs.numpy_input_fn(x={"x": np.array(X_train)}, y=y_train,num_epochs=None,shuffle=False)
test_input_fn = tf.estimator.inputs.numpy_input_fn(x={"x": np.array(X_test)}, y=None,shuffle=False)
classifier.train(input_fn=train_input_fn, steps=100)
preds = classifier.predict(input_fn=test_input_fn)
print(preds)

**Comment by angersson** [comment]
`preds` is indeed a generator object that cannot be printed on its own. [The Estimator Tutorial](https://www.tensorflow.org/get_started/estimator) demonstrates this:

```python
predictions = list(classifier.predict(input_fn=predict_input_fn))
predicted_classes = [p["classes"] for p in predictions]

print(
    "New Samples, Class Predictions:    {}\n"
    .format(predicted_classes))
```

If you have more issues like this, please post a question on [Stack Overflow](http://stackoverflow.com/questions/tagged/tensorflow), which is more suited to individual issues that are not bugs or feature requests. Thanks!

**Comment by XeMinZa** [comment]
I have tried as the tutorial. But it emitted error
`WARNING:tensorflow:Input graph does not contain a QueueRunner. That means predict yields forever. This is probably a mistake.
error: None values not supported.`
so, what's the problem..

**Comment by bluesky314** [comment]
rude

**Comment by luvneries** [comment]
Can we pass batched features from dataset iteration into GBT estimator for prediction instead input_fn?
